src/esp/gfx/RenderCamera.cpp

```diff
@@ -39,10 +43,46 @@ RenderCamera& RenderCamera::setProjectionMatrix(int width,
   return *this;
 }
 
-RenderCamera& RenderCamera::draw(MagnumDrawableGroup& drawables) {
-  // TODO: visibility culling
-  MagnumCamera::draw(drawables);
-  return *this;
+uint32_t RenderCamera::draw(MagnumDrawableGroup& drawables) {
+  uint32_t numDrawnObjects = drawables.size();
+  if (!frustumCulling) {
+    MagnumCamera::draw(drawables);
+  } else {
+    // camera frustum relative to world origin
+    const Mn::Math::Frustum<float> frustum =
+        Mn::Math::Frustum<float>::fromMatrix(projectionMatrix() *
```

Comment:
```suggestion
        Mn::Frustum::fromMatrix(projectionMatrix() *
```

Same elsewhere, and in the test. *All* `Math` classes have convenience float/double typedefs in the root namespace.

Reply:
Good to know. Will change.